src/Vkontakte.php

```diff
@@ -2,184 +2,238 @@
 
 namespace J4k\OAuth2\Client\Provider;
 
-use League\OAuth2\Client\Entity\User;
-use League\OAuth2\Client\Token\AccessToken;
 use League\OAuth2\Client\Provider\AbstractProvider;
-
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Token\AccessToken;
+use Psr\Http\Message\ResponseInterface;
 
 class Vkontakte extends AbstractProvider
 {
-    public $scopes = ['email'];
-    public $uidKey = 'user_id';
-    public $responseType = 'json';
-
-    public function urlAuthorize()
+    protected $baseOAuthUri = 'https://oauth.vk.com';
+    protected $baseUri      = 'https://api.vk.com/method';
+    protected $version      = '5.52';
+
+    /**
+     * @type array
+     * @see https://vk.com/dev/permissions
+     */
+    public $scopes = [
+        'email',
+        'friends',
+        'offline',
+        //'photos',
```

Comment:
Don't leave these commented out like this - remove them if they aren't in use, and allow consumers to pass in additional scopes